src/main/java/cz/smarteon/loxone/app/state/LoxoneState.java

```diff
@@ -0,0 +1,93 @@
+package cz.smarteon.loxone.app.state;
+
+import cz.smarteon.loxone.Loxone;
+import cz.smarteon.loxone.LoxoneAppListener;
+import cz.smarteon.loxone.LoxoneEventListener;
+import cz.smarteon.loxone.LoxoneException;
+import cz.smarteon.loxone.LoxoneProfile;
+import cz.smarteon.loxone.LoxoneUuid;
+import cz.smarteon.loxone.app.AnalogInfoControl;
+import cz.smarteon.loxone.app.Control;
+import cz.smarteon.loxone.app.DigitalInfoControl;
+import cz.smarteon.loxone.app.LoxoneApp;
+import cz.smarteon.loxone.app.SwitchControl;
+import cz.smarteon.loxone.message.TextEvent;
+import cz.smarteon.loxone.message.ValueEvent;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Main entry point of the library if state retention is needed.
+ */
+@Slf4j
+public class LoxoneState implements LoxoneAppListener, LoxoneEventListener {
+
+    private static final Map<Class<? extends Control>,Class<? extends ControlState<? extends Control>>> SUPPORTED_CONTROLS_STATE_MAP;
+
+    private final Loxone loxone;
+
+    private LoxoneApp loxoneApp;
+
+    private Map<LoxoneUuid, ControlState<?>> controlStates;
+
+    static {
+        SUPPORTED_CONTROLS_STATE_MAP = new HashMap<>();
+        SUPPORTED_CONTROLS_STATE_MAP.put(SwitchControl.class, SwitchControlState.class);
+        SUPPORTED_CONTROLS_STATE_MAP.put(AnalogInfoControl.class, AnalogInfoControlState.class);
+        SUPPORTED_CONTROLS_STATE_MAP.put(DigitalInfoControl.class, DigitalInfoControlState.class);
+    }
+
+    public LoxoneState(LoxoneProfile loxoneProfile) {
```

Comment:
well, consider having another constructor taking `Loxone` as parameter - it will ease the testing and also allow user to use single instance of `Loxone` class for both state and other usecases.  

You can keep this constructor for convenience.

Mark parameter nullability and check if needed.